onnx/defs/generator/defs.cc

```diff
@@ -523,4 +528,193 @@ ONNX_OPERATOR_SET_SCHEMA(
+    const TensorProto* delta) {
+
+    if (start->dims().size() != 0 || 
+        limit->dims().size() != 0 ||
+        delta->dims().size() != 0) {
+      fail_shape_inference(
+          "Input to 'Range' op should be scalars (Tensor with only one element and shape empty)");
+    }
+
+  const auto& start_data = ParseData<T>(start);
+  const auto& limit_data = ParseData<T>(limit);
+  const auto& delta_data = ParseData<T>(delta);
+
+  int64_t n = static_cast<int64_t>(
+      ceil((1.0 * (limit_data[0] - start_data[0])) / delta_data[0]));
+
+  if (n < 0)
+    n = 0;
+
+  if (n > static_cast<int64_t>(std::numeric_limits<int>::max()))
+    fail_shape_inference(
+        "Computed output dimension value is outside the bounds "
+        "of the highest value that can be accomodated");
+
+  return static_cast<int>(n);
+}
+
+const std::vector<NodeProto> build_nodes_range_op() {
+    // body for 'Loop node'
+    GraphProto loop_sub_graph;
+    loop_sub_graph.set_name("loop_body_attribute");
+
+    // 'Loop' node 'body' attribute's graph inputs
+    auto* input_value_info_proto_0 = loop_sub_graph.add_input();
+    input_value_info_proto_0->set_name("i");
+
+    auto* input_value_info_proto_1 = loop_sub_graph.add_input();
+    input_value_info_proto_1->set_name("cond");
+
+    auto* input_value_info_proto_2 = loop_sub_graph.add_input();
+    input_value_info_proto_2->set_name("prev");
+
+    // 'Loop' node 'body' attribute's graph nodes
+    auto* node_proto_0 = loop_sub_graph.add_node();
+    node_proto_0->set_op_type("Identity");
+    node_proto_0->add_input();
+    node_proto_0->set_input(0, "cond");
+    node_proto_0->add_output();
+    node_proto_0->set_output(0, "cond_out");
+
+    auto* node_proto_1 = loop_sub_graph.add_node();
+    node_proto_1->set_op_type("Add");
+    node_proto_1->add_input();
+    node_proto_1->set_input(0, "prev");
+    node_proto_1->add_input();    
+    node_proto_1->set_input(1, "delta");
+    node_proto_1->add_output();
+    node_proto_1->set_output(0, "current");
+
+    auto* node_proto_2 = loop_sub_graph.add_node();
+    node_proto_2->set_op_type("Identity");
+    node_proto_2->add_input();
+    node_proto_2->set_input(0, "prev");
+    node_proto_2->add_output();
+    node_proto_2->set_output(0, "range");
+
+    // 'Loop' node 'body' attribute's graph inputs
+    auto* output_value_info_proto_0 = loop_sub_graph.add_output();
+    output_value_info_proto_0->set_name("cond_out");
+
+    auto* output_value_info_proto_1 = loop_sub_graph.add_output();
+    output_value_info_proto_1->set_name("current");
+
+    auto* output_value_info_proto_2 = loop_sub_graph.add_output();
+    output_value_info_proto_2->set_name("range");
+
+    return FunctionBodyHelper::BuildNodes({
+      // nodes: {outputs, op, inputs, attributes}
+      {{"sub_result"}, "Sub", {"limit", "start"}},
+      {{"delta_cast"}, "Cast", {"delta"}, {{"to", static_cast<int64_t>(1)}}},
+      {{"div_result"}, "Div", {"sub_result", "delta_cast"}},
```

Comment:
If I remember right, "Div" expects both inputs to be of same type. So, this would require a cast of "sub_result" also.

Reply:
Fixed, thanks a lot!